### PR TITLE
[#134] Add parent_org field to Experience model and setup validations in controller.

### PR DESCRIPTION
**Pull request by evwhite29**
This branch has two major changes:
- Add the new parent_org attribute to the Experience model so that we can track which organization an experience for a program is created under.
- Setup thorough request payload validations for the Experience resource. These validations exist as a combined effort between conditional checks in the experiences controller and validation checks in the experience model.

### CODE REVIEW

**Comment by evwhite29** [comment]
This merge conflict is easily resolvable, but I'll wait until you guys have had a chance to look over the rest of the changes before resolving it since I'm required to make a new commit and merge development in after resolving it.

File: app/controllers/api/v1/experiences_controller.rb
Comment (dan-crayne) [inline app/controllers/api/v1/experiences_controller.rb:51]: Does this check make it impossible for the super admin to make changes to a user's experience? (This may be what we want, I just don't recall.)

Comment (evwhite29) [reply app/controllers/api/v1/experiences_controller.rb:51]: Correct. I believe this is what we want. If a super admin is unhappy with a users experience, then they can simply set the visibility of the user to false. I don't see the need for them to have that kind of granular control and edit capability.

File: app/controllers/api/v1/experiences_controller.rb
Comment (dcrayne) [inline app/controllers/api/v1/experiences_controller.rb]: Why is this being called? It doesn't seem like we're doing anything with the result; is there some kind of side-effect that I don't know about?

File: app/controllers/api/v1/experiences_controller.rb
Comment (dan-crayne) [inline app/controllers/api/v1/experiences_controller.rb:61]: So I'm a bit confused with these validation functions. Do they work by raising exceptions? (It doesn't look like we use any of the return values.)

File: app/controllers/api/v1/experiences_controller.rb
Comment (dan-crayne) [inline app/controllers/api/v1/experiences_controller.rb:83]: This might help to shed light on my question above about the general strategy of these validations: is the purpose of this call to 'Organization.find()' simply to see if it raises an exception?

File: app/controllers/api/v1/experiences_controller.rb
Comment (evwhite29) [reply app/controllers/api/v1/experiences_controller.rb:61]: Good question. Yes, the main purpose of these validate functions are to check for erroneous conditions that can't be easily recovered from or should not be recovered from. If they pass, then the actual action method (in this case either create or update) will execute normally and confidently pass the params to the model method that may or may not validate further aspects of each param.

File: app/controllers/api/v1/experiences_controller.rb
Comment (evwhite29) [reply app/controllers/api/v1/experiences_controller.rb:83]: Yes, this is the easiest way to have the proper exception be thrown if an organization does not exist for the specified ID. Unfortunately or fortunately. Other existence check methods only return true or false, which I didn't want to have to catch just to throw an exception anyway.

**Review by dan-crayne — APPROVED** [review]
Looks good! I had a few questions about the general strategy of the code, but I don't feel that they warrant holding this one back.

**Review by iandusenbury — APPROVED** [review]

**Comment by iandusenbury** [comment]
will merge once conflicts are resolved

**Pull request merged**
